### PR TITLE
chore: Update influxdb converter to v0.2.16

### LINKED ISSUE #7049: Permanent error: failed to convert OTLP metric to line protocol
Got error when insert spanmetric into **influx** but metric can(data from kafka).

opentelemetry-collector-contrib: v0.40.0

diagram:
`service -> opentelemetry-collector-contrib(metric, spanmetric) -> kafka -> opentelemetry-collector-contrib(consumer) -> influx`

> 2022-01-05T08:52:06.853Z	error	exporterhelper/queued_retry.go:183 Exporting failed. The error is not retryable. Dropping data. {"kind": "exporter", "name": "influxdb/test_historgram", "error": "Permanent error: failed to convert OTLP metric to line protocol: invalid metric histogram bucket counts qty 10 vs explicit bounds qty 10", "dropped_items": 92}

config(from kafka to influx)
```yaml
receivers:
  kafka/metric:
    brokers:
      - localhost:9092
    topic: telemetry.metric.otlp
    group_id: otel-collector-metric-consumer
    auth:
      tls:
        insecure: true
    protocol_version: 2.8.0
  
  kafka/spanmetric:
    brokers:
       - localhost:9092
    topic: telemetry.spanmetric.otlp
    group_id: otel-collector-spanmetric-consumer
    auth:
      tls:
        insecure: true
    protocol_version: 2.8.0


processors:
  batch:
    timeout: 3s

memory_limiter:
  limit_mib: 120
  check_interval: 4s

extensions:
  zpages: {}
  
  memory_ballast:
    # Memory Ballast size should be max 1/3 to 1/2 of memory.
    size_in_percentage: 30
  
  health_check:

exporters:
  prometheus:
    endpoint: "0.0.0.0:8877"
    namespace: "genesis"

  influxdb/test:
    endpoint: http://localhost
    timeout: 5000ms
    sending_queue:
      enabled: true
      num_consumers: 3
      queue_size: 10
    retry_on_failure:
      enabled: true
      initial_interval: 1s
      max_interval: 3s
      max_elapsed_time: 10s
    org: test@test-email
    bucket: test
    token: token
    metrics_schema: telegraf-prometheus-v2

  influxdb/test_historgram:
    endpoint: http://localhost
    timeout: 5000ms
    sending_queue:
      enabled: true
      num_consumers: 3
      queue_size: 10
    retry_on_failure:
      enabled: true
      initial_interval: 1s
      max_interval: 3s
      max_elapsed_time: 10s
    org: test@test-email
    bucket: test_historgram
    token: token
    metrics_schema: telegraf-prometheus-v2

service:
  extensions: [zpages, memory_ballast, health_check]
  pipelines:
    metrics:
      receivers: [kafka/metric]
      processors: [memory_limiter, batch]
      exporters: [influxdb/test] // ok

    metrics/spanmetrics:
      receivers: [kafka/spanmetric]
      processors: [memory_limiter, batch]
      exporters: [influxdb/test_historgram] // got error
```

### PR DESCRIPTION
**Pull request by jacobmarble**
**Description:** <Describe what has changed.>

The [InfluxDB exporter follows](https://github.com/influxdata/influxdb-observability/blob/14db1a7807e294fd3fea47c084818cdb42f07710/otel2influx/metrics_telegraf_prometheus_v2.go#L186-L189) the [OTLP protocol description](https://github.com/open-telemetry/opentelemetry-proto/blob/1648cf3ff58af20e7721c0e95b226c4f3336dee9/opentelemetry/proto/metrics/v1/metrics.proto#L464-L471):

```
  // The number of elements in bucket_counts array must be by one greater than
  // the number of elements in explicit_bounds array.
```

The OTel [Metrics Data Model doc](https://github.com/open-telemetry/opentelemetry-specification/blob/660826e60a9aebad8f97e52e3cb352877c8d8ec7/specification/metrics/datamodel.md#histogram) doesn't add a lot:

>  - (optional) A series of buckets with:
>    - Explicit boundary values.  These values denote the lower and upper bounds
>      for buckets and whether not a given observation would be recorded in this
>      bucket.
>    - A count of the number of observations that fell within this bucket.

However:
- it isn't the exporter's job to enforce convention
- many exporters appear to accept the missing "infinity bucket"
- the Prometheus exporter doesn't even check for the infinity bucket
- the two schemas in the current InfluxDB exporter follows Prometheus convention

Therefore, the InfluxDB exporter should not needlessly reject a histogram that doesn't specify an infinity bucket.

**Link to tracking Issue:** <Issue number if applicable>

Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/7049

**Testing:** <Describe what testing was performed and which tests were added.>

Added unit tests to the updated package in https://github.com/influxdata/influxdb-observability/pull/52

**Documentation:** <Describe the documentation added.>
n/a

### CODE REVIEW

**Review by anuraaga — COMMENTED** [review]
Is the change completely in the external influxdb-observability repo? In that case, I would make the PR title just "Update influxdb converter to v0.2.16" or something since it's unusual to have a bugfix without any code. The actual fix could still be mentioned in the PR description.

Or if not the case, I can't find the business logic changes in this PR

**Comment by pmm-sumo** [comment]
> Is the change completely in the external influxdb-observability repo? In that case, I would make the PR title just "Update influxdb converter to v0.2.16" or something since it's unusual to have a bugfix without any code. The actual fix could still be mentioned in the PR description.
> 
> Or if not the case, I can't find the business logic changes in this PR

That's a good question. We historically included description of changes in the bumped libraries such as opentelemetry-log-collection ([example](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8124/files)), which is used for `filelog` receiver. The caveat was that those changes were in libraries directly used in the collector (almost like components) and in fact implemented with collector in mind. I think this is also the case here. If I would be using InfluxDB that information would be helpful (since changelog is one of the first resources when I'm searching if a possible problem has been already solved with a newer version)

**Review by pmm-sumo — APPROVED** [review]
LGTM overall, there are several `go.sum` though which I think should be not part of this PR

File: internal/coreinternal/go.mod [hunk 1]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/stretchr/testify v1.7.0
 	go.opentelemetry.io/collector v0.46.1-0.20220307173244-f980c9ef25b1
 	go.opentelemetry.io/collector/model v0.46.1-0.20220307173244-f980c9ef25b1
+	go.uber.org/zap v1.21.0
```
Comment (pmm-sumo) [inline internal/coreinternal/go.mod:13]: Why this (minor) change? (and similar once in attributesprocessor)

**Comment by jacobmarble** [comment]
> If I would be using InfluxDB that information would be helpful (since changelog is one of the first resources when I'm searching if a possible problem has been already solved with a newer version)

Not sure we have consensus. I've updated the PR description, but left the changelog entry as-is.

> there are several go.sum though which I think should be not part of this PR

This is the result of `make gotidy && go mod tidy`. I've reverted that and rebased on a recent checkout of main.

File: internal/coreinternal/go.mod (hunk 1 above)
Comment (jacobmarble) [reply internal/coreinternal/go.mod]: Fixed.

**Review by bogdandrutu — APPROVED** [review]

**Comment by codeboten** [comment]
Closing and reopening to trigger CI

**Review by codeboten — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Pull request merged**